drivers/include/saul/periph.h

```diff
@@ -53,6 +57,20 @@ typedef struct {
 } saul_adc_params_t;
 #endif /* MODULE_SAUL_ADC */
 
+#ifdef MODULE_SAUL_PWM
+/**
+ * @brief   Direct mapped PWM configuration values
+ */
+typedef struct {
+    const char *name;       /**< name of the device connected to this pin */
+    pwm_t line;             /**< PWM line to initialize and expose */
+    pwm_mode_t mode;        /**< PWM mode */
+    uint32_t freq;          /**< PWM frequency */
+    uint16_t res;           /**< PWM resolution */
+    uint8_t channel;        /**< PWM channel */
+} saul_pwm_params_t;
```

Comment:
This struct contains a lot of redundant information. The configuration should reflect the structure of the PWM devices -> n devices with x channels each.